application-services/custom/camera-management/appcamera/commands.go

```diff
@@ -76,42 +66,64 @@ func (app *CameraManagementApp) doPTZ(deviceName, profileToken string, x, y, zoo
 	return app.sendPutCommand(deviceName, relativeMoveCommand, cmd)
 }
 
-func (app *CameraManagementApp) getPresets(deviceName string, profileToken string) (GetPresetsResponse, error) {
-	cmd := &ptz.GetPresets{
-		ProfileToken: onvif.ReferenceToken(profileToken),
-	}
-
-	presets, err := app.issueGetCommand(context.Background(), deviceName, getPresetsCommand, cmd)
+func (app *CameraManagementApp) getCameraFeatures(deviceName string) (CameraFeatures, error) {
+	var err error
+	features := CameraFeatures{}
+	resp, err := app.service.DeviceClient().DeviceByName(context.Background(), deviceName)
 	if err != nil {
-		return GetPresetsResponse{}, errors.Wrapf(err, "failed to issue get presets command")
+		return CameraFeatures{}, err
 	}
 
-	val := presets.Event.Readings[0].ObjectValue
-	js, err := json.Marshal(val)
-	if err != nil {
-		return GetPresetsResponse{}, errors.Wrapf(err, "failed to marshal presets json object")
+	switch resp.Device.ServiceName {
+	case app.config.AppCustom.OnvifDeviceServiceName:
+		features.CameraType = Onvif
+		caps, err := app.getCapabilities(deviceName)
+		if err != nil {
+			return CameraFeatures{}, errors.Wrapf(err, "unable to get device capabilities")
+		}
+		if caps.Capabilities.PTZ.XAddr != "" {
+			features.PTZ = true
+
+			ptzConfigs, err := app.getPTZConfiguration(deviceName)
+			if err == nil {
```

Comment:
error not nil also needs to be handled here.

Reply:
yeah, i guess i should at least log it. do you think we should disable PTZ if this call fails or not?

Reply:
you mean disable zoom if it fails?

Reply:
@presatish , yes, basically set `features.PTZ` to false if the get configuration fails.